[JavaScript] Create definition for let keyword (#95)
exercism/v3: languages/javascript/keywords/let.md

# `let`

> The `let` keyword declares and initializes a new variable. It can be optionally defined to a value.

Reviewer: ```suggestion

The `let` keyword declares and initializes a new variable. It can be optionally defined to a value.
```

Add newline

Reviewer: Did you mean:

```suggestion

The `let` keyword declares a block scope local variable, optionally initializing it to a value.
```

Author: Does
```javascript
let foo;
```
not initialize the `foo` variable?

Reviewer: It initialises it to `undefined`. The problem is this:

> It can be optionally defined to a value

```javascript
let foo = 'x'
```

Defines `foo`, and initialises (assigns for the first time) a string literal of `x`. It doesn't optionally define to a value. The wording is just slightly off :)

```javascript
let arrow;
arrow = "->";

// or

let arrow = "->";
```

> Variables declared using the `let` keyword have scope in the block in which they are defined, and any contained sub-blocks. The main difference between `let` and [`var`](https://github.com/exercism/v3/blob/master/languages/javascript/keywords/var.md) is that the scope of `var` is the entire function.

Reviewer: Link to variable, scope and use relative link for `var`.

Reviewer: > is the entire function.

I don't think that's right. I think you mean when there is an enclosed function.

```javascript
function bob() {

>   console.log(name); // Variable name is undefined

Reviewer: Can we use an example that doesn't rely on hoisting?

  {
    let name = "Bob";
  }
}
```

```javascript
function steve() {
  console.log(name); // Variable name is defined, as "Steve"
  {
    var name = "Steve";
  }
}
```

> # `let`

Reviewer: Double content
